### PR TITLE
Enable enumerating ES6 class properties as additional options

### PR DESCRIPTION
**Pull request by antitoxic**
This takes in account when options are being passed as ES6 class. Related to: https://github.com/skatejs/skatejs/pull/450

### CODE REVIEW

**Comment by treshugart** [comment]
Unfortunately using `for of` requires a `Symbol` polyfill which I don't think we should be using at this stage (see build). You can try using `for in` with `hasOwnProperty()` checks, but I had issues with only doing own properties and ES6 classes at one point (don't recall the exact issue). Maybe it's ok now.

File: src/index.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ function makeOptions (userOptions) {
   }
 
   // Copy over non-standard options.
-  for (let name in userOptions) {
+  for (let name of Object.getOwnPropertyNames(userOptions)) {
```
Comment (treshugart) [inline src/index.js]: You might be able to `assign(options, userOptions)` (via `import assign from 'object-assign') here instead which may fix the CI (might cause other issues we don't know about).

Comment (antitoxic) [inline src/index.js]: I saw `object-assign` a bit later on. I will switch to it for this.

**Comment by antitoxic** [comment]
:christmas_tree: commit with `assign-safe` jingle while watching "Home Alone" for the 100th time.

**Comment by treshugart** [comment]
@antitoxic I wonder if we should just implement https://github.com/skatejs/skatejs/issues/447. I don't think this is copying over props because ones inherited from classes aren't "own".

**Comment by treshugart** [comment]
Tried fixing in #462. Maybe we can write a test for this in that one?

**Comment by treshugart** [comment]
Hey @antitoxic, just reading this in relation to #450, which is now merged, I'm not sure what exact problem this is solving. Could you elaborate? If I read this correctly, it's making class properties enumerable when they're not by default. It sounds ok to me to leave them as is, unless it's solving a use-case. Thoughts?

**Comment by antitoxic** [comment]
Probably not solving much. I'm simply mimicking the behavior that skate has when it's passed a plain javascript object as options.

Without this patch, behaviors differ. It doesn't bring any problems right now. However a inconsistency might bring problems later on. The #450 is an example of one such inconsistency. #450 is not a bug for passing options as plain object but it is when passing options as class (function).

**Comment by treshugart** [comment]
I wonder if #450 is still an issue in master. I think this behaviour is perfectly fine. You could pass a normal object with enumerable props and you'd get the same behaviour as a class. Going to close, but I'd be happy to re-open if there is a use-case or bug related to this.

**Comment by antitoxic** [comment]
If this pull request or something similar is not merged the small problem that might not be obvious but will remain for the future is that  if skate options were changed but "default options" were not plain object settings will work but class setttings won't. Just to keep in mind.

**Comment by treshugart** [comment]
Having defaults out of sync with the API is a bug, so those should always be correct as your other PR corrected :)

**Comment by antitoxic** [comment]
That begs the question why is than this needed:

```
// Copy over non-standard options.
for (let name in userOptions) {
  options[name] = userOptions[name];
}
```

at all?

**Comment by treshugart** [comment]
It's not in master anymore. This is what the file looks like now: https://github.com/skatejs/skatejs/blob/master/src/index.js#L61.

**Comment by antitoxic** [comment]
:+1: My bad. That's cool.

**Pull request closed without merging**